CMakeLists.txt

```diff
@@ -70,27 +70,79 @@ list(REMOVE_ITEM my_project_headers
 		 ${PROJECT_SOURCE_DIR}/Analysis/include/QwSIS3320_Samples.h
 		 )
 
+
 #----------------------------------------------------------------------------
-# MYSQLPP
+# sqlpp11
 #
-find_package(MYSQLPP)
-IF(MYSQLPP_FOUND)
-  config_add_dependency(MYSQLPP)
-  include_directories(${MYSQLPP_INCLUDE_DIR})
-  ## KLUDGE:  For now, do not define the __USE_DATABASE__ flag, because we're still missing some of the database support files. --- P.King, 05jun2018
-  message(STATUS "  The \"__USE_DATABASE__\" flag is commented out in CMakeLists.txt; database support is DISABLED.")
-  #  add_definitions(-D__USE_DATABASE__)
-ELSE(MYSQLPP_FOUND)
-  set(MYSQLPP_LIBRARIES "")
+list(APPEND SQLPP11_OPTIONAL_COMPONENTS SQLite3 MySQL MariaDB PostgreSQL)
+find_package(Sqlpp11
```

Comment:
Explore CMake alternatives for compilation: FetchContent, subproject.

Reply:
With 3073f24909ba65cd6c6f2d7743fdcaa7b4490b53 there are now three modalities:
- sqlpp11 available through find_package (after dedicated build or system-wide install),
- sqlpp11 available in thirdparty/sqlpp11 in a git submodule clone,
- no sqlpp11, no database functionality.

This maintains backwards compatibility, doesn't add additional compilation burden for those who don't want database functionality, and allows centralized dependency management and control over version used to interact with the database.

Reply:
Seems there are still some unresolved issues since the backend dependencies are not propagated.